File: internal/config/wizard_profiles.go
package config

// WizardTrack identifies first-run setup paths in the desktop wizard.
type WizardTrack string

const (
	WizardTrackDeveloper    WizardTrack = "developer"
	WizardTrackLifeSciences WizardTrack = "lifeSciences"
	WizardTrackGeneral      WizardTrack = "general"
)

// WizardProfile holds model and agent defaults for a setup track.
type WizardProfile struct {
	Track            WizardTrack
	OllamaModel      string
	ModelsToEnsure   []string
	HFHostedModel    string // Hub repo id for huggingface provider
	CloudAnthropic   string
	DefaultAgents    []AgentConfig
}

const (
	// BioOllamaChatModel is the recommended local OpenBio chat model (Ollama hub; Llama 3 template).
	BioOllamaChatModel = "koesn/llama3-openbiollm-8b:latest"
	// BioOllamaToolModel runs MCP biology tools when the chat model lacks native tool calling.
	BioOllamaToolModel = "qwen2.5:7b"
	// UtilityOllamaModel is the hub utility tier for session summaries and similar background tasks.
	UtilityOllamaModel = "qwen2.5:7b"
	// BioOllamaTag is the canonical Ollama model name for OpenBioLLM (import via HF GGUF).
	BioOllamaTag = "nj-bio:8b"
	// BioHFRepo is the Hugging Face instruct model for hosted inference.
	BioHFRepo = "aaditya/Llama3-OpenBioLLM-8B"
	// BioHFGGUFRepo is the GGUF catalog repo for local import.
	BioHFGGUFRepo = "aaditya/OpenBioLLM-Llama3-8B-GGUF"
)

// WizardProfileFor returns defaults for developer, life sciences, or general setup.
func WizardProfileFor(track WizardTrack) WizardProfile {
	switch track {
	case WizardTrackLifeSciences:
		return WizardProfile{
			Track:          WizardTrackLifeSciences,
			OllamaModel:    BioOllamaChatModel,
			ModelsToEnsure: []string{BioOllamaChatModel, BioOllamaToolModel},
			HFHostedModel:  BioHFRepo,
			CloudAnthropic: "claude-3-5-sonnet-20241022",
			DefaultAgents: []AgentConfig{
				{Type: "biology", Name: "BiologyExpert", Enabled: true},
				{Type: "assistant", Name: "Assistant", Enabled: true},
			},
		}
	case WizardTrackGeneral:
		return WizardProfile{
			Track:          WizardTrackGeneral,
			OllamaModel:    UtilityOllamaModel,
			ModelsToEnsure: []string{UtilityOllamaModel},
			HFHostedModel:  "",
			CloudAnthropic: "claude-3-5-sonnet-20241022",
			DefaultAgents: []AgentConfig{
				{Type: "assistant", Name: "Assistant", Enabled: true},
			},
		}
	default:
		return WizardProfile{
			Track:          WizardTrackDeveloper,
			OllamaModel:    DevOllamaCodeModel,
			ModelsToEnsure: []string{DevOllamaCodeModel, UtilityOllamaModel},
			HFHostedModel:  "",
			CloudAnthropic: "claude-3-5-sonnet-20241022",
			DefaultAgents: []AgentConfig{
				{Type: "assistant", Name: "Assistant", Enabled: true},
			},
		}
	}
}

// ApplyWizardProfile mutates cfg for a completed wizard track (Ollama local path).
func (c *Config) ApplyWizardProfile(track WizardTrack, ollamaLocal bool) {
	p := WizardProfileFor(track)
	c.Agents = append([]AgentConfig(nil), p.DefaultAgents...)
	if c.Packs.Enabled == nil {
		c.Packs = DefaultPacksConfig()
	}
	switch track {
	case WizardTrackLifeSciences:
		c.Packs.Enabled[PackLifeSciences] = true
		c.Packs.Enabled[PackSoftwareDevelopment] = false
	case WizardTrackDeveloper:
		c.Packs.Enabled[PackLifeSciences] = false
		c.Packs.Enabled[PackSoftwareDevelopment] = true
	default:
		c.Packs.Enabled[PackLifeSciences] = false
		c.Packs.Enabled[PackSoftwareDevelopment] = false
	}
	c.SyncAgentsFromPacks()
	c.Ollama.ModelsToEnsure = append([]string(nil), c.Ollama.ModelsToEnsure...)

	if ollamaLocal {
		c.AI.DefaultProviderID = "ollama-local"
		for i := range c.AI.Providers {
			if c.AI.Providers[i].ID == "ollama-local" {
				c.AI.Providers[i].Model = p.OllamaModel
				return
			}
		}
		c.AI.Providers = []ProviderConfig{{
			ID:       "ollama-local",
			Type:     "ollama",
			Name:     "Local Ollama",
			Endpoint: "http://localhost:11434",
			Model:    p.OllamaModel,
		}}
	}
}

File: desktop/src/config/wizardProfiles.ts
/** Keep in sync with internal/config/wizard_profiles.go */

export type WizardTrack = 'developer' | 'lifeSciences' | 'cad' | 'general';

/** Primary OpenBio chat model (Ollama Hub). Keep in sync with internal/config/wizard_profiles.go */
export const BIO_OLLAMA_CHAT_MODEL = 'koesn/llama3-openbiollm-8b:latest';
export const BIO_OLLAMA_TOOL_MODEL = 'qwen3.5:9b';
/** Optional HF GGUF import tag */
export const BIO_OLLAMA_TAG = 'nj-bio:8b';
export const DEV_OLLAMA_MODEL = 'qwen3.5:27b';
export const CAD_OLLAMA_CHAT_MODEL = 'qwen3.5:27b';
export const CAD_OLLAMA_CHAT_MODEL_LIGHT = 'qwen3.5:9b';
export const CAD_OLLAMA_TOOL_MODEL = 'qwen3.5:9b';
export const CAD_OLLAMA_TAG = 'nj-cad:27b';
export const UTILITY_OLLAMA_MODEL = 'qwen3.5:9b';

export interface WizardAgentChoice {
  type: string;
  name: string;
  enabled: boolean;
}

export function agentsForTrack(track: WizardTrack): WizardAgentChoice[] {
  if (track === 'lifeSciences') {
    return [
      { type: 'biology', name: 'BiologyExpert', enabled: true },
      { type: 'assistant', name: 'Assistant', enabled: true },
    ];
  }
  if (track === 'cad') {
    return [
      { type: 'cad', name: 'CADExpert', enabled: true },
      { type: 'assistant', name: 'Assistant', enabled: true },
    ];
  }
  if (track === 'general') {
    return [{ type: 'assistant', name: 'Assistant', enabled: true }];
  }
  return [
    { type: 'assistant', name: 'Assistant', enabled: true },
    { type: 'backend', name: 'BackendEngineer', enabled: true },
  ];
}

export function ollamaModelForTrack(track: WizardTrack): string {
  if (track === 'lifeSciences') return BIO_OLLAMA_CHAT_MODEL;
  if (track === 'cad') return CAD_OLLAMA_CHAT_MODEL;
  if (track === 'general') return UTILITY_OLLAMA_MODEL;
  return UTILITY_OLLAMA_MODEL;
}

export function modelsToEnsureForTrack(track: WizardTrack, providerType: 'ollama' | 'cloud'): string[] {
  if (providerType !== 'ollama') return [];
  if (track === 'lifeSciences') return [BIO_OLLAMA_CHAT_MODEL, BIO_OLLAMA_TOOL_MODEL];
  if (track === 'cad') return [CAD_OLLAMA_CHAT_MODEL, CAD_OLLAMA_CHAT_MODEL_LIGHT, CAD_OLLAMA_TOOL_MODEL];
  if (track === 'general') return [UTILITY_OLLAMA_MODEL];
  return [UTILITY_OLLAMA_MODEL];
}

export function packsEnabledForTrack(track: WizardTrack): Record<string, boolean> {
  return {
    ide: track === 'developer',
    'life-sciences': track === 'lifeSciences',
    'software-development': track === 'developer',
    cad: track === 'cad',
  };
}

/** Infer focus track from hub packs.enabled (re-run setup prefill). */
export function inferWizardTrackFromPacks(
  enabled: Record<string, boolean> | null | undefined,
): WizardTrack {
  if (!enabled) return 'general';
  if (enabled['life-sciences']) return 'lifeSciences';
  if (enabled.cad) return 'cad';
  if (enabled.ide || enabled['software-development']) return 'developer';
  return 'general';
}
